crypto/mem.c

```diff
@@ -162,6 +162,20 @@ void ossl_malloc_setup_failures(void)
 }
 #endif
 
+static void *report_malloc_failure(void *result, const char *file)
+{
+    if (result == NULL && file != NULL) {
+        int len = strlen(file);
+
+        /* prevent error-in-error-handler malloc loop */
+        if (len < 6
+            ? strcmp(file, "err.c") != 0
+            : strcmp(file + len - 6, "/err.c") != 0)
+            ERR_raise(ERR_LIB_CRYPTO, ERR_R_MALLOC_FAILURE);
```

Comment:
Shouldn't the original _file_ and _line_ be reported here?

We could then take out a lot of MALLOC_FAILURE messages elsewhere.

Reply:
> Shouldn't the original _file_ and _line_ be reported here?

Good point! Done.

> We could then take out a lot of MALLOC_FAILURE messages elsewhere.

Yes indeed :smile: